reference/3.0/Microsoft.PowerShell.Core/About/about_Ref.md

```diff
@@ -5,72 +5,156 @@ locale:  en-us
 keywords:  powershell,cmdlet
 title:  about_Ref
 ---
-# About Ref
-
-## SHORT DESCRIPTION
 
-Describes how to create and use a reference variable type.
+# About Ref
 
-## LONG DESCRIPTION
+## Short description
 
-You can use the reference variable type to permit a method to change the value
+Describes how to create and use a reference type variable. You can use
+reference type variables to permit a method to change the value
 of a variable that is passed to it.
 
-When the [ref] type is associated with an object, it returns a reference to
-that object. If the reference is used with a method, the method can refer to
-the object that was passed to it. If the object is changed within the method,
-the change appears as a change in the value of the variable when control
-returns to the calling method.
+## Long description
 
-To use referencing, the parameter must be a reference variable. If it is not,
-an InvalidArgument exception is thrown.
+You can pass variables to methods *by reference* or *by value*.
 
-The parameters used in method invocations must match the type required by the
-methods.
+In the following example, a method attempts to change the value of a
+variable that is passed into it, and fails.
 
-Examples:
+When you pass a variable *by value*, you are passing a copy of the data.
 
 ```powershell
-PS> function swap([ref]$a,[ref]$b)
->> {
->>     $a.value,$b.value = $b.value,$a.value
->> }
-
-PS> $a = 1
-PS> $b = 10
-PS> $a,$b
-1
-10
-PS> swap ([ref]$a) ([ref]$b)
-PS> $a,$b
+Function Test($data)
+{
+    $data = 3
+}
+
+$var = 10
+Test -data $var
+$var
+```
+
+```output
 10
-1
+```
+
+In the following example, a variable containing a `Hashtable` is passed to a
+method. `Hashtable` is an object type so by default it is passed to the
+method *by reference*.
+
+When passing a variable *by reference*, the method can change the data and
+that change will persist after the method executes.
+
+```powershell
+Function Test($data)
+{
+    $data.Test = "New Text"
+}
+
+$var = @{}
+Test -data $var
+$var
+```
+
+```output
+Name                           Value
+----                           -----
+Test                           New Text
+```
+
+The method adds its own key, which you can still view after the
+method executes.
```

Comment:
The function adds a new key-value pair that persists outside of the function's scope.

Reply:
check